posts/2020/08/the-ibis-backends.md

```diff
@@ -0,0 +1,359 @@
+    </tr>
+    <tr>
+      <th>2018</th>
+      <td>31</td>
+      <td></td>
+      <td>10</td>
+      <td>71</td>
+      <td>35</td>
+      <td>8</td>
+      <td>17</td>
+      <td></td>
+      <td></td>
+      <td>9</td>
+      <td>2</td>
+      <td>2</td>
+    </tr>
+    <tr>
+      <th>2019</th>
+      <td>33</td>
+      <td>22</td>
+      <td>17</td>
+      <td>12</td>
+      <td>32</td>
+      <td>1</td>
+      <td>4</td>
+      <td></td>
+      <td>7</td>
+      <td>1</td>
+      <td>2</td>
+      <td>5</td>
+    </tr>
+    <tr>
+      <th>2020</th>
+      <td>38</td>
+      <td>3</td>
+      <td>4</td>
+      <td>2</td>
+      <td>4</td>
+      <td>1</td>
+      <td>2</td>
+      <td>1</td>
+      <td>3</td>
+      <td>4</td>
+      <td>4</td>
+      <td></td>
+    </tr>
+  </tbody>ibis direct execution backends like pandas and hdf5 operate on conventional in-memory python objects. pandas is the gold standard for structured data in python, and inspires the API for ibis.
+
+
+</table>
+</div>
+<br>
+
+> We note an early focus `ibis.sqlite`, `sqlalchemy` and `ibis.impala`. 
+Later, work began on the `pandas` backend rounding out the three different types of backgrounds.
+From this point, improvements were made to these key backends as `ibis.clickhouse`, `ibis.spark` and `ibis.postgres`. 
+For the past 3 years, Quansight, in partnership with OmniSci, added the `ibis.omniscidb`
+string generating backend. Further, our responsibilities have expanded
+to support `ibis` as community maintainers through Quansight Labs. 
+This collaboration introduced geospatial functionality to `ibis` for several backends.
+
+### Currently, there is an ongoing effort to add sqlserver backends
+
+There are on going efforts to introduce [SQL Server][sql-server] support.  
+What other backends would you like to see for `ibis`?
+Maybe our community could benefit from a `dask` backend or `altair` backend?
+
+## `ibis` direct execution backends
+
+`ibis` direct execution backends like `pandas` and `hdf5` operate on conventional in-memory python objects.
+`pandas` is the gold standard for structured data in python, and inspires the API for `ibis`.
+
+
+```python
+    pd = ibis.pandas.connect({'A': pandas.util.testing.makeDataFrame()})
+```
+
+`pd` is an `ibis` backend based off `pandibis direct execution backends like pandas and hdf5 operate on conventional in-memory python objects. pandas is the gold standard for structured data in python, and inspires the API for ibis.
+
+as.DataFrame` objects.
+
+```python
+    expression = pd.table('A').head()
+```
+
+`expression` is an `ibis` query, that has `expression.compile` and `expression.execute` methods.
+We'll recognize the __execute__ method when we return `pandas.DataFrame`s from `ibis` expression.
+The __compile__ method does not trigger any computation, rather it constructs an intermediate form
+that is interpretted by a backend.
+
+```python
+    >>> assert isinstance(expression.compile(), ibis.expr.types.TableExpr)
```

Comment:
There's some Python code blocks with `>>>` and some without - this looks a little odd, could you make it consistent?

Reply:
Also, deindenting all code blocks would be good, some look off

Reply:
should i remove the doctest syntax. i user them in there to indicate a thing is a test. in my development environment i'm running tests against them.

happy to make changes based on your preferences. I'll dedent the code blocks.

Reply:
It's fine to keep, but then all lines that you execute should have `>>>`. That'd be best here, since otherwise you cannot easily distinguish input from output.